contracts/src/Rolldown.sol

```diff
@@ -400,6 +406,8 @@ contract Rolldown is
                 withdrawal.withdrawalRecipient,
                 pendingEthWithdrawals[withdrawal.withdrawalRecipient]
             );
+            //TODO:: remove with protocol update
```

Comment:
Don't you need to add to pendingEthWithdrawals if the code.length is non-zero?
Coz this way the funds will be lost if the recipient is a contract...

Reply:
the faild status is brought back to the L2 so we can return funds if withdrawals has failed. buttttttttttt we are about to change the protocol either way, so lets take a closer look then, and for testnet we can live with things as they are i guess